judoscale-ruby/lib/judoscale/utilization_tracker.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "singleton"
+
+module Judoscale
+  class UtilizationTracker
+    include Singleton
+
+    def initialize
+      @mutex = Mutex.new
+      @active_request_counter = 0
+      @started = false
+    end
+
+    def start!
+      @mutex.synchronize do
+        unless started?
+          @started = true
+          init_idle_report_cycle!
+        end
+      end
+    end
+
+    def started?
+      @started
+    end
+
+    def incr
+      @mutex.synchronize do
+        if @active_request_counter == 0 && @idle_started_at
+          # We were idle and now we're not - add to total idle time
+          @total_idle_time += get_current_time - @idle_started_at
+          @idle_started_at = nil
+        end
+
+        @active_request_counter += 1
+      end
+    end
+
+    def decr
+      @mutex.synchronize do
+        @active_request_counter -= 1
+
+        if @active_request_counter == 0
+          # We're now idle - start tracking idle time
+          @idle_started_at = get_current_time
+        end
+      end
+    end
+
+    def idle_ratio(reset: true)
```

Comment:
Have you considered swapping this to busy_ratio like we briefly mentioned the other day?